src/js/relevant.js

```diff
@@ -226,19 +308,175 @@ export default {
         );
     },
 
+    /**
+     * @typedef ToggleNonRelevantModleNodesOptions
+     * @property {number} [repeatIndex]
+     * @property {number} [repeatPath]
+     * @property {boolean} setRelevant
+     */
+
+    /**
+     * @typedef {import('./nodeset').Nodeset} NodeSet
+     */
+
+    /**
+     * @typedef RepeatInfo
+     * @property {number} repeatIndex
+     * @property {string} repeatPath
+     */
+
+    /**
+     * @param {HTMLElement} branchNode
+     * @param {string} path
+     * @param {ToggleNonRelevantModleNodesOptions} options
+     */
+    toggleNonRelevantModelNodes(branchNode, path, options) {
```

Comment:
This is the bulk of it, challenge was balancing correctness and performance

TODO: H come back